src/Nox.Lib/Application/Providers/DefaultSystemProvider.cs

```diff
@@ -1,9 +1,24 @@
-﻿using Nox.Abstractions;
+﻿using Microsoft.AspNetCore.Http;
+using Nox.Abstractions;
 using Nox.Types;
 
 namespace Nox.Application.Providers;
 
 public class DefaultSystemProvider : ISystemProvider
 {
-    public string GetSystem() => "N/A";
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public DefaultSystemProvider(IHttpContextAccessor httpContextAccessor) =>
+        _httpContextAccessor = httpContextAccessor;
+
+    /// <summary>
+    /// Get the system name from the X-System-Name header
+    /// </summary>
+    /// <returns></returns>
+    public string GetSystem()
+    {
+        var result = _httpContextAccessor.HttpContext?.Request.Headers["X-System-Name"].ToString();
+        result ??= string.Empty;
```

Comment:
Not sure what would be better, Empty? "NotFound"? Not Available? @gjorgji-grgovski-authoritypartners any suggestion?
```suggestion
        result ??= string.Empty;
```

Reply:
The Header is supposed to be populated by the Kong Gateway, ideally, we would not have it as an empty, but in that case, I would still keep it as an empty string.
The thing here is that I think we should keep these DefaultProviders and create new ones like HeaderSystemProvider and HeaderUserProvider.

Reply:
If not found, returns "N/A" as we decided